src/components/3d_editor/advanced_geometry/python_transformation/PythonCodeExecution.tsx

```diff
@@ -0,0 +1,50 @@
+import IconByName from "@exabyte-io/cove.js/dist/mui/components/icon/IconByName";
+import theme from "@exabyte-io/cove.js/dist/theme";
+import CheckIcon from "@mui/icons-material/Check";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import CircularProgress from "@mui/material/CircularProgress";
+import Typography from "@mui/material/Typography";
+import React from "react";
+
+interface PythonExecutionControlsProps {
+    isLoading: boolean;
+    isRunning: boolean;
+    handleRun: () => void;
+}
+
+function PythonCodeExecution(props: PythonExecutionControlsProps) {
+    const { isLoading, isRunning, handleRun } = props;
+
+    const getStatusText = () => {
+        if (isLoading) return "Loading...";
+        if (isRunning) return "Running...";
+        return "Ready";
+    };
+
+    return (
+        <Box sx={{ display: "flex", justifyContent: "flex-end" }} gap={1}>
+            <Box sx={{ display: "flex", alignItems: "center" }}>
+                <Typography variant="body2">{getStatusText()}</Typography>
+                {isLoading || isRunning ? (
+                    <CircularProgress color="primary" size={theme.typography.button.fontSize} />
+                ) : (
+                    <CheckIcon color="success" />
+                )}
+            </Box>
+            <Button
+                id="python-transformation-dialog-run-button"
```

Comment:
lets use `className="run-button"` and update the test selector to `run-button`. The wrapper selector is already defined